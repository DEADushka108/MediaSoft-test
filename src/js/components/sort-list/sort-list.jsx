import React, {useState} from 'react';
import PropTypes from 'prop-types';
import {ActionCreator} from '../../store/ships/ships';
import {connect} from 'react-redux';
import {getActiveSort} from '../../store/ships/selectors';

const SortList = (props) => {
  const {currentSort, onSortClick} = props;
  const [activeSort, setActiveSort] = useState(currentSort);

  return <ul className="catalog__sort-list">
    <li className={`catalog__sort-item ${(activeSort === `asc`) ? `catalog__sort-item--active` : ``}`} onClick={() => {
      setActiveSort(`asc`);
      onSortClick(`asc`);
    }}>
      <a className="catalog__sort-link">
        <svg className="catalog__sort-icon">
          <use xlinkHref="#sort-up"></use>
        </svg>
        <span className="visually-hidden">Ascending</span>
      </a>
    </li>
    <li className={`catalog__sort-item ${(activeSort === `des`) ? `catalog__sort-item--active` : ``}`} onClick={() => {
      setActiveSort(`des`);
      onSortClick(`des`);
    }}>
      <a className="catalog__sort-link">
        <svg className="catalog__sort-icon">
          <use xlinkHref="#sort-down"></use>
        </svg>
        <span className="visually-hidden">Descending</span>
      </a>
    </li>
    <li className="catalog__sort-item" onClick={() => {
      setActiveSort(``);
      onSortClick(``);
    }}>
      <a className="catalog__sort-link">
        <svg className="catalog__sort-icon">
          <use xlinkHref="#refresh"></use>
        </svg>
        <span className="visually-hidden">Refresh</span>
      </a>
    </li>
  </ul>;
};

SortList.propTypes = {
  currentSort: PropTypes.string,
  onSortClick: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  currentSort: getActiveSort(state),
});

const mapDispatchToProps = (dispatch) => ({
  onSortClick(sort) {
    dispatch(ActionCreator.setActiveSort(sort));
  }
});

export {SortList};
export default connect(mapStateToProps, mapDispatchToProps)(SortList);
