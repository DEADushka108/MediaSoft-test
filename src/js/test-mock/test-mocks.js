export * from './const/const';
export * from './ships/ships';
export * from './store/store';
