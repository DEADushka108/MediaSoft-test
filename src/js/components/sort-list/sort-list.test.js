import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {store, noop} from '../../test-mock/test-mocks';
import {SortList} from './sort-list';


it(`Should SortList component render correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <SortList
          currentSort={``}
          onSortClick={noop}
        />
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
