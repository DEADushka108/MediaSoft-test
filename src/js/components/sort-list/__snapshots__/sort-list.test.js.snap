// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should SortList component render correctly 1`] = `
<ul
  className="catalog__sort-list"
>
  <li
    className="catalog__sort-item "
    onClick={[Function]}
  >
    <a
      className="catalog__sort-link"
    >
      <svg
        className="catalog__sort-icon"
      >
        <use
          xlinkHref="#sort-up"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        Ascending
      </span>
    </a>
  </li>
  <li
    className="catalog__sort-item "
    onClick={[Function]}
  >
    <a
      className="catalog__sort-link"
    >
      <svg
        className="catalog__sort-icon"
      >
        <use
          xlinkHref="#sort-down"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        Descending
      </span>
    </a>
  </li>
  <li
    className="catalog__sort-item"
    onClick={[Function]}
  >
    <a
      className="catalog__sort-link"
    >
      <svg
        className="catalog__sort-icon"
      >
        <use
          xlinkHref="#refresh"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        Refresh
      </span>
    </a>
  </li>
</ul>
`;
